postprocessor/src/main/java/com/azure/autorest/postprocessor/util/PartialUpdateHandler.java

```diff
@@ -0,0 +1,135 @@
+package com.azure.autorest.postprocessor.util;
+
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.NodeList;
+import com.github.javaparser.ast.body.BodyDeclaration;
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.ast.body.TypeDeclaration;
+import com.github.javaparser.ast.expr.SimpleName;
+import com.github.javaparser.ast.nodeTypes.NodeWithSimpleName;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.github.javaparser.StaticJavaParser.parse;
+
+public class PartialUpdateHandler {
+
+    public static String handlePartialUpdateForFile(String generatedFileContent, String existingFileContent) {
+        CompilationUnit compilationUnitForGeneratedFile = parse(generatedFileContent);
+        CompilationUnit compilationUnitForExistingFile = parse(existingFileContent);
+        ClassOrInterfaceDeclaration generatedClazz = getClassOrInterfaceDeclaration(compilationUnitForGeneratedFile);
+        ClassOrInterfaceDeclaration existingClazz = getClassOrInterfaceDeclaration(compilationUnitForExistingFile);
+
+        List<BodyDeclaration<?>> generatedFileMembers = new ArrayList<>();
+        if (generatedClazz != null) {
+            generatedFileMembers = generatedClazz.getMembers();
+        }
+        List<BodyDeclaration<?>> existingFileMembers = new ArrayList<>();
+        if (existingClazz != null) {
+            existingFileMembers = existingClazz.getMembers();
+        }
+
+        // 1. check if the file is in scope of partial update:
+        // if there is a method has @Generated annotation, then the file is in scope of partial update, otherwise return directly
+        boolean hasGeneratedAnnotations = generatedFileMembers.stream().anyMatch(member -> hasGeneratedAnnotation(member));
+
+        if (!hasGeneratedAnnotations) {
+            return generatedFileContent;
+        }
+
+        NodeList<BodyDeclaration<?>> updatedMembersList = new NodeList<>();
+        // 2. iterate existingFileMembers, keep manual written members, and replace generated members with the corresponding newly generated one
+        for (BodyDeclaration<?> existingMember : existingFileMembers) {
+            boolean isGeneratedMethod = hasGeneratedAnnotation(existingMember);
+            if (!isGeneratedMethod) { // manual written member
+                updatedMembersList.add(existingMember);
```

Comment:
We should call this list `manuallyAddedMembersList` or something similar to make it explicit that these are the members that were not generated.

Reply:
Because `updatedMemberList` is the member list that we will set to `generatedClazz`, 
```java
generatedClazz.setMembers(updatedMembersList);
``` 
It not only contains manually added member, but also contains generated members, 'Code Change' section in description explains more details about `updatedMemberList`